Update documentation for redis.io (#967)
RedisGears/RedisGears: docs/docs/concepts/triggers/User_Functions.md

---
title: "User Functions"
linkTitle: "User Functions"
weight: 1
description: >
    Execute JavaScript functions via `TFCALL` or `TFCALLASYNC`
---

> The arguments given on the [`TFCALL`](docs/commands.md#tfcall) command, after the function name, will be passed to the function callback. The following example shows how to implement a simple function that returns the value of a key whether its a string or a hash:

Reviewer: ```suggestion
All [`TFCALL`](docs/commands.md#tfcall) command that follow the function name are passed to the function callback. The following example shows how to implement a simple function that returns the value of a key of type string or hash:
```

```js
#!js api_version=1.0 name=lib

redis.registerFunction('my_get', function(client, key_name){
    if (client.call('type', key_name) == 'string') {
        return client.call('get', key_name);
    }
    if (client.call('type', key_name) == 'hash') {
        return client.call('hgetall', key_name);
    }
    throw "Unsupported type";
});
```

> Example of running the following function:

Reviewer: ```suggestion
Example of running the preceding function:
```

```bash
127.0.0.1:6379> set x 1
OK
127.0.0.1:6379> TFCALL lib.my_get 1 x
"1"
127.0.0.1:6379> hset h a b x y
(integer) 2
127.0.0.1:6379> TFCALL lib.my_get 1 h
1) "a"
2) "b"
3) "x"
4) "y"

```

> It is also possible to get all th arguments given to the function as a `JS` array. This is how we can extend the above example to except multiple keys and return their values:

Reviewer: ```suggestion
It is also possible to get all the function arguments as a `JS` array. This is how we can extend the above example to accept multiple keys and return their values:
```
